Prepare for new toolbar

WouterJ opened the pull request:
Prepare for the new toolbar design: symfony/symfony#15160

![sf-sonata-block-toolbar](https://cloud.githubusercontent.com/assets/749025/9008686/455f0ca8-379c-11e5-8ba7-3b44a2e02f81.png)

I created the SVG from the PNG logo, if you have a proper SVG file, it should be used instead.

Soullivaneuh commented:
Is that BC with older toolbar too?

WouterJ commented:
@Soullivaneuh yes, the `profiler_markup_version` variable defines the version of the toolbar. The older toolbar has no such variable, so it's set to 1 (using the `default` filter), the upcoming version of the toolbar has version 2.

Using this variable, you can support different versions of the toolbar. E.g. when version 3 comes out, you have:

    {% if profiler_markup_version == 1 %}
        {# 2.7- #}
    {% elseif profiler_markup_version  == 2 %}
        {# 2.8 #}
    {% else %}
        {# latest #}
    {% endif %}

Soullivaneuh commented:
Okay nice idea. But just major versions?

Soullivaneuh commented:
Note: Will review this when symfony/symfony#15160 will be merged.

WouterJ commented:
@Soullivaneuh toolbar markup version is not bound to Symfony's version. If "3", it just means it's the 3th design of the toolbar (that can already happen in 3.1 for instance).

Soullivaneuh commented:
I'm not talking about Symfony version but semantic versioning itself. ;-)

WouterJ commented:
Ah, in that case: Yes :) (there aren't going to be much redesigns that require markup changes).

Soullivaneuh commented:
![sonata_block_debug_bar](https://cloud.githubusercontent.com/assets/1698357/9036707/201afd36-39e4-11e5-94ff-88f47dede888.png)

Two things:

* Sonata logo seems to be cropped.
* Maybe we could try to get the same style like other section. Logo + number square. What do you think?

Soullivaneuh commented:
And for the second point, maybe we could put a more explicit icon that Sonata logo.

Like this one for example: http://fontawesome.io/icon/cubes/

ping @rande.

rande commented:
@WouterJ yep you can change the icon as @Soullivaneuh suggest.

Soullivaneuh commented:
@WouterJ SVG format could be found here: https://github.com/FortAwesome/Font-Awesome/blob/master/fonts/fontawesome-webfont.svg

rande commented:
ok to test

rande commented:
ok to test

rande commented:
ok to test

WouterJ commented:
I like the suggestion of changing the icon to something more describing. I changed it to another icon than the one suggested, as I find the style of this icon more consistent with the other icons and it looks better when it's just 24x24. This is also the icon used on the sonata-project.org website.

Btw, @Soullivaneuh, the official sonata project logo *is* cropped: https://sonata-project.org/bundles/sonatageneral/images/logo.png

rande commented:
Ok for the logo

Soullivaneuh commented:
> Btw, @Soullivaneuh, the official sonata project logo is cropped: https://sonata-project.org/bundles/sonatageneral/images/logo.png

Yes, and it's not adapted for the toolbar IMHO.

OK for the logo too, but I would like you to remove "blocks" and setup number rendering as other components:

![sonata_block_icons](https://cloud.githubusercontent.com/assets/1698357/9168625/b8c5a2ba-3f58-11e5-9dbd-e59565297429.png)

Like doctrine does.

Soullivaneuh commented on Resources/views/Profiler/block.html.twig:
```diff
@@ -1,7 +1,10 @@
+
     <div class="sf-toolbar-block">
+        {% if profiler_markup_version == 1 %}
         <div class="sf-toolbar-icon">
```
Should be indented.

Soullivaneuh commented on Resources/views/Profiler/block.html.twig:
```diff
@@ -9,6 +12,16 @@
             </a>
         </div>
+        {% else %}
+        <a href="{{ path('_profiler', { 'token': token, 'panel': name }) }}">
```
Should be indented.

Soullivaneuh commented on Resources/views/Profiler/block.html.twig:
```diff
@@ -18,12 +31,16 @@
                 <span>{{ collector.containers|length }}</span>
             </div>
+            {% if profiler_markup_version == 1 %}
+            {# don't show the total number of blocks in the info in the new design,
```
Should be indented.

WouterJ commented:
Thanks @Soullivaneuh, I've fixed the remaining issues.

Please note that the number shown is already equal to the other numbers. The doctrine panel, as shown in each screenshot in this PR, uses the old style. You can see the new styles here: https://github.com/symfony/symfony/pull/15160 I've removed "blocks" though.

Soullivaneuh commented:
Thank you @WouterJ.

WouterJ commented:
Thanks @Soullivaneuh for your review & merge!
